### DISCUSSION TITLE
Improvements on Samsung Z Fold 4?

### DISCUSSION THREAD

**Discussion by parapenT1sta** [NightscoutFoundation/xDrip#3196] [General]
I am using a Samsung Z Fold 4 and the status line covers most of the graph when using the inner screen. Is it possible to improve it?

Cover screen:
![Screenshot_20231119_113425_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/608650/6d925dc8-1152-4882-8f02-47635ef7939d)

Inner screen:
![Screenshot_20231119_113437_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/608650/3500da03-8cc2-4734-9e07-72c82a824212)

**Comment by Navid200** [comment]
I don't have a Samsung.
Someone who does needs to recreate the problem and if it is something in xDrip, it can be fixed.

**Comment by Navid200** [comment]
The phone may have a setting related to newer versions of Android to provide customizations that cause this.

  
Is there a feature on the phone that allows customization of the font size in each app?
There are new features of Android that we may not have added to xDrip.  We need to keep xDrip compatible with older versions of Android to avoid forcing everyone to rush to the store to buy a new phone.  There are countries where even a low-cost phone can be as expensive of the monthly income of a family.

**Reply by parapenT1sta** [reply]
There is no customization for each screen. I wish it had.
Is anything else I can try?
I think my issue also affects new Fold models.

**Reply by Navid200** [reply]
So, your concern is the inner screen not the cover display, right?

Can we talk in this thread instead of the other?  It never helps to mix issues brought up by different people.
Can you post both screen resolutions here in this thread?

**Reply by Navid200** [reply]
xDrip detects screen size and adjusts the text size based on that.
But, this is not done with a great level of resolution.  It is quite possible that the resolution of the inner screen in this case makes xDrip jump up to a larger font too soon.  So, we may need to add an additional step to increase resolution.

But, I need more info from you to confirm that.
I can also give you a test release to verify before it can be changed.

**Reply by parapenT1sta** [reply]
> So, your concern is the inner screen not the cover display, right?
> 
> Can we talk in this thread instead of the other? It never helps to mix issues brought up by different people. Can you post both screen resolutions here in this thread?

That's correct. Only the inner screen.

Only "time ago", "last reading" and "status line" lines are big, the rest information for me it's with the perfect size.

<img width="1812" height="502" alt="Screenshot_20250728_161513_Brave" src="https://github.com/user-attachments/assets/19f68f00-fc29-4d1e-a0be-d9fdacabe126" />

**Reply by parapenT1sta** [reply]
Do you need my email to test or do you share it here?

**Reply by Navid200** [reply]
No need for email

**Reply by Navid200** [reply]
I made a virtual device with that screen resolution.  Does this look right?  
<img width="678" height="819" alt="Screenshot 2025-07-28 162726" src="https://github.com/user-attachments/assets/1fcb1d21-1c50-4d7b-b08e-f3cb5538555b" />

**Reply by parapenT1sta** [reply]
Yes. If you put more info on the status line you will see that will fill up a lot of the screen.

**Reply by Navid200** [reply]
No, I'm not debating that.  The challenge now is how to fix this in such a way that it would get approved.
This may take a while.  It may take a few days if not a week.

**Reply by parapenT1sta** [reply]
@Navid200 Thank you a lot for all your effort trying to fix it. Take your time.

**Reply by Navid200** [reply]
The device is being detected as a tablet.
If you rotate your phone unfolded to go into landscape mode, you will see that it does not look too bad.  It looks like a very small tablet.
I'm saying it now makes sense what's going on.  In the past, a device with these characteristics only existed as a tablet and tablets are used in landscape.  So, I understand why xDrip did this in the past.

Now, we have fold devices that are being detected as tablets.  This is a problem.  So, I still believe we need to fix this.

**Reply by parapenT1sta** [reply]
The font size it's the same but looks better. I was curious and I found out that in fact Google mentioned about apps detecting fold devices:

https://android-developers.googleblog.com/2023/06/detecting-if-device-is-foldable-tablet.html?m=1

https://developer.android.com/develop/ui/compose/layouts/adaptive/foldables/make-your-app-fold-aware

**Comment by Navid200** [comment]
Bu the way, you know that you can remove those time buttons from the main screen, right?

**Reply by parapenT1sta** [reply]
Yes, I know. The same menu I setup the arrow. The time buttons don't bother me... for now. 😅

**Reply by Navid200** [reply]
OK,

Any approach I look at results in a problem looking at all devices.
The behavior of xDrip is intentional so that on a tablet that is meant to be kept farther from your face, the fonts remain readable.
When you unfold your phone, it effectively becomes like a tablet.

Don't all the other apps behave the same way on your phone in that sense?

**Reply by parapenT1sta** [reply]
Not all apps have the same behavior, but for example, Gmail app, reading an email on the outter screen the letter size is smaller compared to inner screen when I open it to read the same email but the difference of the letter size is nothing compared to difference in xDrip app.

**Reply by Navid200** [reply]
Thanks for stating that.
That means that Gmail also changes the font size when you use the inner screen.

A decision was made a long time ago how to scale the fonts going from a phone to a tablet.  Changing that today will impose a change on everyone using xDrip.  This does not mean you are wrong.  This means maintaining an app is not as easy as it may seem.

If we were creating an app today, perhaps we would do many things differently than what is done in xDrip.  But, we are not creating a new app.
Can we add a setting to let the user make a choice?  We can always do that.  That's why we are were we are today with so many settings in xDrip half of which no one knows they even exist.  I am not a fan of adding new settings.  

xDrip has a very coarse display binning.  Your phone, when unfolded has a screen size of 7.6".  This is not an extremely large tablet.   One option is to add more grids to xDrip for screen size binning.  Then, the fonts will not become so much larger on your phone.  Again, this will affect many existing users.  I am almost certain such a submission will be rejected exactly for that reason.

Another option is to add a detection mechanism to figure out if the device is a foldable device or not.  If it is, use a different scaling for it to make sure a foldable like yours will not have very large fonts.  The problem with that is that some time in the future, someone with your type of phone will complain why the font size is so small on their screen!

I know Android lets the user select a font size.
Does your phone let you choose two fonts sizes one for unfolded and another for folded?
If yes, can you use those two to make sure you are happy with both?
I know this means your Gmail font will become a little bit smaller.  Can you give it a try?

I really would like to find a solution.  But, I believe, if you were in my place, you might also have more questions than answers like me.

**Reply by parapenT1sta** [reply]
@Navid200 For the first time I installed xDrip on my Doogee T30PRO 11"  tablet with 1600x2560 resolution and I also felt the font size was also big for this screen, the same feeling I have with inner screen of my fold device. My question is also, how many people use xDrip on their tablet VS phone?

I am sharing screen shots taken from an email viewing from outter, inner screen and tablet and screenshots from xDrip on tablet.

I have no option to define font size for each screen.

Outter screen:
<img width="970" height="2347" alt="Screenshot_20250730_215721_Gmail" src="https://github.com/user-attachments/assets/736aea25-b470-4d9f-b498-f2ac85234408" />

Inner screen:
<img width="1812" height="2049" alt="Screenshot_20250730_215732_Gmail" src="https://github.com/user-attachments/assets/85d7a6c7-7162-412e-a66c-1548d25bcf14" />

Tablet:
<img width="1600" height="2560" alt="Screenshot_20250730-220030" src="https://github.com/user-attachments/assets/0ace6b1b-54d5-4424-baca-1fe4e69647a0" />

<img width="1600" height="2560" alt="Screenshot_20250730-220222" src="https://github.com/user-attachments/assets/d8502b1b-8d93-4d7d-b715-e2e0b5eaf1ad" />

<img width="1600" height="2560" alt="Screenshot_20250730-220417" src="https://github.com/user-attachments/assets/24bfec1d-5c73-44b5-9880-7a0f55c9d7de" />

**Reply by Navid200** [reply]
I have no way of knowing how many people use what.  xDrip does not collect any data from users.
What we see in forums represents what people who participate in forums do.  By no means does it represent all the xDrip users.  
We are still keeping Android 7 coverage included.  How many people are still using Android 7?  Same answer.  We have absolutely no idea.

**Reply by parapenT1sta** [reply]
In my option font size should be reduce a little bit for higher resolution screens as tablet and fold inner screens. Maybe no one complaint before because they don't use xdrip with tablets and when it was coded the tablets had lower resolution.

You need to click on the pictures to view the real difference between them.

**Reply by parapenT1sta** [reply]
@Navid200 Sorry for repeating myself, but now that I am also comparing with a tablet I feel the change will affect all devices in a good way.

Can you just reduce a little bit the font not much and I compare it in both devices?

**Reply by Navid200** [reply]
It will be rejected.  
I would like to make changes to xDrip to improve it.  Many of such intents have been rejected because it would have resulted in a change in existing behavior.

If @jamorham approves, I will make the change and reduce the size of the fonts for tablets a little bit.
But, without that, it is pointless to work on a submission knowing ahead of time that there is 90% chance it will be rejected.

**Reply by Navid200** [reply]
I can detect that the phone is foldable and I can detect that the device is being detected as a tablet.
If those two are true, I can reduce the amount of font size magnification.  But, even foldables are not brand new.  The first foldable Samsung came out in 2019.
So, why has no one else complained about this?

**Reply by parapenT1sta** [reply]
Maybe because not many have a fold device or really don't report or care about the font size. Fold devices are not cheap.
Like me, I have a tablet and never installed xDrip before until today to compare. We will get so many answers to the same questions.

**Reply by Navid200** [reply]
When I look at my phone to see xDrip, it is usually for seeing what my glucose is or for snoozing an alert.  There are also times that I am starting a sensor or calibrating.
But, I cannot think of a scenario where I would be sitting there and watching xDrip on my phone like a movie.

Can you help me understand when exactly you will be unfolding your phone in order to look at xDrip instead of just dealing with it on the cover without unfolding the phone?

**Reply by parapenT1sta** [reply]
I understand what you are saying but as I said before I also think the font letter is big on tablets. For example if you compare how the status line increase from the outter screen resolution to inner/tablet resolution is way more than the glucose value on the top left. If you compare the increase you will see that the increase is really huge. What I am saying is that the status line doesn't have the same pattern of the glucose and that is why it stands out.

My question is why status line increases and the glucose value doesn't? When the most important value on the screen is the glucose and not the status line, at least for me.

And I would like you again to compare the increase with status line and glucose value... see how small it is on outter screen and then on the inner screen/tablet gets the same attention as the glucose value just because of the size... that's why it bothers me. It's not important when compare to glucose. Can you understand at least? 

I think I gave all my points of view of someone who has a fold device and a tablet. I don't have more valid points. For me this topic is ended. End of discussion for me since I have no more valid arguments for it.

Thank you for this discussion.

Keep the good work.

**Comment by Navid200** [comment]
For anyone experiencing this problem who may find this thread through search:

As of the 2025.09.19 release, there is a new settings t:
Settings > Display Settings > Font Settings > Enlarge fonts on larger displays

If you disable this setting, xDrip will not automatically enlarge the fonts.  This should reduce the font size for you.
